components/docked-composer/index.jsx

```diff
@@ -0,0 +1,168 @@
+/* Copyright (c) 2015-present, salesforce.com, inc. All rights reserved */
+/* Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license */
+
+// Implements the [Docked Composer design pattern](https://lightningdesignsystem.com/components/docked-composer/) in React.
+// Based on SLDS v2.9.4
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+import shortid from 'shortid';
+
+import { DOCKED_COMPOSER } from '../../utilities/constants';
+
+import Button from '../button';
+
+const propTypes = {
+	actionButton: PropTypes.node,
+	composerBody: PropTypes.node,
+	header: PropTypes.shape({
+		icon: PropTypes.node,
+		label: PropTypes.string,
+	}),
+	id: PropTypes.string,
+	isOpen: PropTypes.bool,
+};
+
+class DockedComposer extends React.Component {
```

Comment:
The overflow menu would take a Dropdown `overflowMenuDropdown` and have a number label. `labels.overflowPanelNumber`